.github/workflows/on-merge-queue.yml

```diff
@@ -0,0 +1,65 @@
+---
+# Merge queue.  A PR that reaches here has already passed the more rigorous PR
+# suite, so is very likely to pass.  The main reasons for failures here are
+# flaky VMs timing out (which we can't do much about), or a merge conflict
+# with another PR that is also in the merge queue.
+#
+# There's no reason to have multiple stages in this case, because we're
+# expecting it to pass.  Having more than one stage frustrates parallel
+# throughput in low-contention cases, and guarantees a longer critical path.
+name: Merge queue
+on:
+  merge_group:
+    branches: [main, "stable/*"]
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  docs_lint:
+    if: github.repository_owner == 'Qiskit'
+    name: Docs and Lint
+    uses: ./.github/workflows/lint_docs.yml
+  test-linux:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/test-linux.yml
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.13"]
+        runner: ["ubuntu-latest", "ubuntu-24.04-arm"]
+    with:
+      python-version: ${{ matrix.python-version }}
+      install-optionals: ${{ matrix.python-version == '3.9' }}
+      runner: ${{ matrix.runner }}
+  test-rust:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/rust-tests.yml
+    with:
+      python-version: "3.9"
+      runner: "ubuntu-latest"
+  test-images:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/image-tests.yml
+    with:
+      python-version: "3.9"
+      runner: "ubuntu-latest"
+  test-mac:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/test-mac.yml
+    strategy:
+      matrix:
+        runner: ["macos-13", "macos-14"]
+    with:
+      python-version: "3.13"
+      install-optionals: false
+      runner: ${{ matrix.runner }}
+  test-windows:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/test-windows.yml
+    strategy:
+      matrix:
+        runner: ["windows-latest"]
```

Comment:
Using a matrix here seems a bit unnecessary since there isn't more than 1 configuration but it's fine for now. I can see a world where we add windows arm support in several years so lets leave the flexibility for the future.